src/final/ArduCAM.cpp

```diff
@@ -124,11 +124,28 @@ ArduCAM::ArduCAM() {
 
 void ArduCAM::begin() {
 	// [TODO] set up CS GPIO pin, (unnecessary if spi is configured with hardware slave management)
+	
+	// set to alternate function mode (10)
+	GPIOB->MODER &= ~(GPIO_MODER_MODER12_0); // PB12
+	//set AF to pin PB12
+	GPIOB->AFR[1] &= ~(GPIO_AFRH_AFSEL12);
+	
+	//af number= 5
+	GPIOB->AFR[1] |= (5 << (12*4));  
```

Comment:
```suggestion
	
	//af number= 5
	GPIOB->AFR[1] |= (5 << (12*4));  
```
No longer necessary since the pin won't be used as AF mode.